lib/packExternalModules.js

```diff
@@ -193,22 +192,49 @@ module.exports = {
       // (1.a) Install all needed modules
       const compositeModulePath = path.join(this.webpackOutputPath, 'dependencies');
       const compositePackageJson = path.join(compositeModulePath, 'package.json');
-      this.serverless.utils.writeFileSync(compositePackageJson, '{}');
 
-      this.serverless.cli.log('Packing external modules: ' + compositeModules.join(', '));
+      // (1.a.1) Create a package.json
+      const compositePackage = {
+        name: this.serverless.service.service,
+        version: '1.0.0',
+        description: `Packaged externals for ${this.serverless.service.service}`,
+        private: true
+      };
+      _.forEach(compositeModules, compositeModule => {
+        const splitModule = _.split(compositeModule, '@');
+        // If we have a scoped module we have to re-add the @
+        if (_.startsWith(compositeModule, '@')) {
+          splitModule.splice(0, 1);
+          splitModule[0] = '@' + splitModule[0];
+        }
+        const moduleVersion = _.join(_.tail(splitModule), '@');
+        _.set(compositePackage, `dependencies.${_.first(splitModule)}`, moduleVersion);
+      });
+      this.serverless.utils.writeFileSync(compositePackageJson, JSON.stringify(compositePackage, null, 2));
 
-      return new BbPromise((resolve, reject) => {
+      // (1.a.2) Copy package-lock.json if it exists, to prevent unwanted upgrades
+      const packageLockPath = path.join(path.dirname(packageJsonPath), 'package-lock.json');
+      return BbPromise.fromCallback(cb => fse.pathExists(packageLockPath, cb))
```

Comment:
Why not use the sync versions if you're using it in a synchronous way?

Reply:
Because it will stall the Node event loop (even if something else might be running in the background - outside of the function) - we still run in the Serverless context and should not affect our host in any way ;-)

Reply:
Tell that to the Serverless guys using synchronous functions whenever humanly possible 🤣

Reply:
Haha... In the meantime they even turned some of the stuff asynchronously, like file copy.